habitat-lab/habitat/datasets/rearrange/samplers/receptacle.py

```diff
@@ -57,6 +58,15 @@ def __init__(
         self.parent_object_handle = parent_object_handle
         self.parent_link = parent_link
 
+        # The unique name of this Receptacle instance in the current scene.
+        # This name is a combination of the object instance name and Receptacle name.
+        self.unique_name = ""
+        if self.parent_object_handle is None:
+            # this is a stage receptacle
+            self.unique_name = "stage|" + self.name
```

Comment:
Is this one guaranteed unique ? This assumes self.name is unique for receptacles without parents.

Reply:
Great point. It "should" be because these are pulled from the stage config rather than a set of object configs. Maybe it is best to add an assertion for uniqueness of stage Receptacle names in the loading phase to ensure this is true.

Reply:
I added this check.